### PR TITLE
W8 - D1 - BaoLe - REST API with Retrofit and Gson

### PR DESCRIPTION
**Pull request by at-baole**
#43

### CODE REVIEW

File: app/src/at-baole/java/asiantech/internship/summer/restapi/RestAPIActivity.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        }
+    }
+
+    private void choosePhotosFromGallery() {
+        Intent pickIntent = new Intent(Intent.ACTION_PICK,
+                android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+        startActivityForResult(pickIntent, GALLERY);
+    }
+
+    private void CapturePhotosFromCamera() {
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/restapi/RestAPIActivity.java]: ERROR: [MethodNamingConventions] Method names should not start with capital letters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#MethodNamingConventions

File: app/src/at-baole/java/asiantech/internship/summer/restapi/ListImageAdapter.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package asiantech.internship.summer.restapi;
+
+import android.content.Context;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/restapi/ListImageAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.content.Context'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/restapi/ListImageAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.content.Context'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-baole/java/asiantech/internship/summer/restapi/RestAPIActivity.java [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/restapi/RestAPIActivity.java]: ERROR: [MethodNamingConventions] Method names should not start with capital letters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#MethodNamingConventions

File: app/src/at-baole/java/asiantech/internship/summer/restapi/RestAPIActivity.java [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+                    mImages.add(0, response.body());
+                    mImageAdapter.notifyItemInserted(0);
+                    mProgressDialog.dismiss();
+                    Toast.makeText(RestAPIActivity.this, getString(R.string.uploadSuccess), Toast.LENGTH_SHORT).show();
+                }
+            }
+
+            @Override
+            public void onFailure(@NonNull Call<ImageItem> call, @NonNull Throwable t) {
+                Toast.makeText(RestAPIActivity.this, getString(R.string.uploadFailed), Toast.LENGTH_SHORT).show();
```
Comment (at-tainguyen) [inline app/src/at-baole/java/asiantech/internship/summer/restapi/RestAPIActivity.java]: Missing `mProgressDialog.dismiss();` for this case

**Comment by at-tainguyen** [comment]
LGTM!

**Pull request merged**